### PR TITLE
CONSOLE-3643: Add edit annotation modal

### PR DESCRIPTION
**Pull request by jhadvig**

### CODE REVIEW

File: integration-tests/e2e/list-page.cy.ts [hunk 1]
```diff
@@ -40,6 +41,33 @@ describe(`${PLUGIN_NAME} list page test`, () => {
     cy.get(".pf-c-alert.pf-m-inline.pf-m-danger").should("not.exist");
     cy.byLegacyTestID("resource-title").should("contain", "my-new-cron-object");
 
+
+
+
+    cy.log("Edit annotations pn the new CronTab from the list page");
```
Comment (rhamilto) [inline integration-tests/e2e/list-page.cy.ts:47]:
```suggestion
    cy.log("Edit annotations on the new CronTab from the list page");
```

Comment (rhamilto) [reply integration-tests/e2e/list-page.cy.ts:47]: Nit:  here and below just 1 blank line.

**Review by rhamilto — COMMENTED** [review]
I know you're still working, but noticed these things.

**Comment by jhadvig** [comment]
Screen of passing tests 
Screenshot 2023-09-22 at 15 56 29

**Pull request merged**